docs/reference/api.md

```diff
@@ -0,0 +1,214 @@
+
+The API uses HTTP over a Unix socket, with access to the API controlled by user ID. If `pebble run` is started with the `--http <address>` option, Pebble exposes a limited set of open-access endpoints (see {ref}`api-access-levels`) using the given TCP address.
+
+## Structure of the API
+
+Here is a breakdown of the general structure of most API endpoints in Pebble.
+
+### API endpoint URL
+
+Pebble API uses standard HTTP URLs to get actions and resources. The URLs are made up of a base URL followed by a path and optional query parameters.
+
+- Base URL: The consistent foundation of all API endpoints (for example, `http://localhost:4000/v1/`).
+- Path: Specifies the resource or collection being accessed (for example, `/changes`, `/services`). Paths are hierarchical and use forward slashes as separators.
+- Query parameters: Optional key-value pairs appended to the URL after a question mark (?) and separated by ampersands (&). They provide additional filtering or control over the request (for example, `/services?name=svc1`).
+
+Example:
+
+```bash
+http://localhost:4000/v1/services?name=svc1
+```
+
+### Request
+
+Key components of a request include:
+
+- HTTP Method: The desired action (for example, `GET` for retrieving data, `POST` for creating data, `PUT` for updating data, and `DELETE` for deleting data).
+- Headers: Metadata about the request, such as content type, basic auth, and accepted response formats (for example, `Authorization: Basic <credentials>`, `Content-Type: application/json`).
+- Body: For requests that send data to the server (`POST`, `PUT`), the body contains the data in JSON format.
+
+### Response
+
+After processing a request, Pebble sends back a response. Key components of a response are:
+
+- HTTP status code: the result of the request.
+- Headers: metadata about the response, such as content type and caching information.
+- Body: data returned by Pebble server in JSON format.
+
+### JSON format
+
+Pebble API uses JSON for data exchange. Both request and response bodies are formatted as JSON objects or arrays.
+
+Example JSON Response:
+
+```json
+{
+  "type": "sync",
+  "status-code": 200,
+  "status": "OK",
+  "result": {
+    "name": "svc1",
+    "startup": "disabled",
+    "current": "stopped"
+  }
+}
+```
+
+### Request types
+
+Pebble API supports both synchronous and asynchronous requests.
+
+- Synchronous requests: The client sends a request and waits for the server to return a response before continuing.
+- Asynchronous requests: For long-running operations, Pebble API offers asynchronous processing. The client sends a request and gets an acknowledgment immediately, but the actual processing is still running on the server side. The client can poll for the result. Specific endpoints supporting asynchronous operations are documented below with `type` as `async`.
+
+## Using cURL
```

Comment:
I think "cURL" is styled as "curl" now, although I also thought it was supposed to be "cURL" at first...